modules/common-fs2/src/main/scala/com/snowplowanalytics/snowplow/enrich/common/fs2/Environment.scala

```diff
@@ -279,7 +282,21 @@ object Environment {
     if (preparedRemoteAdapters.nonEmpty)
       Clients
         .mkHttp(remoteAdapters.connectionTimeout, remoteAdapters.readTimeout, remoteAdapters.maxConnections, ec)
+        .map(enableRemoteAdapterMetrics(metrics))
         .map(c => (c.some, preparedRemoteAdapters))
     else Resource.pure[F, (Option[Http4sClient[F]], Map[(String, String), RemoteAdapter])]((None, preparedRemoteAdapters))
   }
+
+  def enableRemoteAdapterMetrics[F[_]: Sync](metrics: Metrics[F])(inner: Http4sClient[F]): Http4sClient[F] =
+    Http4sClient { req =>
+      for {
+        resp <- inner.run(req).attemptTap {
+                  case Left(_: TimeoutException) => Resource.eval(metrics.remoteAdaptersTimeoutCount)
+                  case Left(_) => Resource.eval(metrics.remoteAdaptersFailureCount)
+                  case Right(response) =>
+                    if (response.status.responseClass == Status.Successful) Resource.eval(metrics.remoteAdaptersSuccessCount)
```

Comment:
The problem with the `==` operator is it's not very type-safe.  Whenever I use `==` I worry that I will make a coding mistake and the compiler won't help me out by telling me.  You might find it interesting to read [this description of the problem from Cats](https://typelevel.org/cats/typeclasses/eq.html).

In this case I would probably have used a `match`, which I know is type safe so I trust the compiler will tell me if I get it wrong

```
    response.status.responseClass match {
      case Status.Successful => Resource.eval(metrics.remoteAdaptersSuccessCount)
      case _ => Resource.eval(metrics.remoteAdaptersFailureCount)
    }
```